File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
# def solution(n):
#     # 에라토스테네스의 체 이용하자!! 가장 효율적
#     a = [False,False] + [True]*(n-1)
#     primes=[]
#     for i in range(2,n+1):
#         if a[i]:
#             primes.append(i)
#             for j in range(2*i, n+1, i):
#                 a[j] = False
    
#     return len(primes)

def solution(n):
    answer = 0
    for i in range(2, n+1):
        for j in range(2, int(i**0.5)+1):
            if i % j == 0:
                break
        else:
            answer += 1
    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
const solution = n => {
    let answer = 0
    const numbers = Array(n+1).fill(0)
    for (let i=2; i<=n; i++) {
        numbers[i] = i
    }
    for (let i=2; i<=n; i++) {
        if (numbers[i] === 0) continue
        for (let j=i+i; j<=n; j+=i) {
            numbers[j] = 0
        }
    }
    for (let i=2; i<=n; i++) {
        if (numbers[i]) answer++
    }
    return answer
}